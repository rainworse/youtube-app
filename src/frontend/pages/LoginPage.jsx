import { useContext } from 'react';
import '../styles/LoginPage.css';
import { useNavigate } from 'react-router-dom';
import UserContext from '../UserContext';
import LocalStorageDB from '../../backend/LocalStorageDB';

const LoginPage = () => {
  const navigate = useNavigate();
  const { setCurrentUser } = useContext(UserContext);

  const onLogin = (e) => {
    e.preventDefault();
    const userName = e.target.elements.username.value;
    setCurrentUser(userName);
    LocalStorageDB.storeUserSession(userName);
    navigate(-1);
  };

  return (
    <div className="login-page">
      <form onSubmit={onLogin}>
        <div className="login-page-inputs">
          <input
            type="text"
            id="username"
            name="username"
            placeholder="username"
          />
          <input type="submit" value="Login" id="login-button" />
        </div>
      </form>
    </div>
  );
};

export default LoginPage;
