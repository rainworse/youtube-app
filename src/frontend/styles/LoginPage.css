.login-page {
  margin-top: 150px;
  display: flex;
  justify-content: center;
  align-items: center;
}

#username {
  font-size: 16px;
  font-family: 'Tinos', serif;
  padding: 8px;
  padding-right: 60px;
  border: none;
  border-radius: 15px;
  border-bottom: 2px solid rgb(243, 243, 243);
  margin-right: -50px;
}

#login-button {
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  background-color: white;
  font-family: 'Tinos', serif;
  font-weight: bold;
}
