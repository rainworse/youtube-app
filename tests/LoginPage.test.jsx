import { render } from '@testing-library/react';
import { describe, expect } from 'vitest';
import { MemoryRouter } from 'react-router-dom';
import UserContext from '../src/frontend/UserContext';
import LoginPage from '../src/frontend/pages/LoginPage';

describe('LoginPage', () => {
  it('renders the login page', () => {
    const loginPage = render(
      <UserContext.Provider value={{ currentUser: 'user' }}>
        <MemoryRouter>
          <LoginPage />
        </MemoryRouter>
      </UserContext.Provider>
    );
    expect(loginPage.container).toMatchSnapshot();
  });
});
