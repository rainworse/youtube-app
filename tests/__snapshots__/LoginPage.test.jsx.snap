// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > renders the login page 1`] = `
<div>
  <div
    class="login-page"
  >
    <form>
      <div
        class="login-page-inputs"
      >
        <input
          id="username"
          name="username"
          placeholder="username"
          type="text"
        />
        <input
          id="login-button"
          type="submit"
          value="Login"
        />
      </div>
    </form>
  </div>
</div>
`;
